### ISSUE TITLE
data loading issue in Libtorch C++

### ISSUE THREAD

**Issue by dhaji** [pytorch/pytorch#13131]
## 🐛 Bug

I am trying to use libtorch in c++ to train the mnist data set. I am ok with cmake part and it downloads the Mnist data set. However, when I issue "make" command I get the following error:

error: ‘torch::data’ has not been declared
 struct Normalize : public torch::data::transforms::TensorTransform<> {

Here is the result of cmake:

cmake -DCMAKE_PREFIX_PATH=~/mnist/libtorch
-- The C compiler identification is GNU 7.3.0
-- The CXX compiler identification is GNU 7.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Downloading MNIST dataset
Downloading http://yann.lecun.com/exdb/mnist/train-images-idx3-ubyte.gz ...
0% |################################################################| 100%
Unzipped /home/davood/mnist/build/data/train-images-idx3-ubyte.gz ...
Downloading http://yann.lecun.com/exdb/mnist/train-labels-idx1-ubyte.gz ...
0% |################################################################| 100%
Unzipped /home/davood/mnist/build/data/train-labels-idx1-ubyte.gz ...
Downloading http://yann.lecun.com/exdb/mnist/t10k-images-idx3-ubyte.gz ...
0% |################################################################| 100%
Unzipped /home/davood/mnist/build/data/t10k-images-idx3-ubyte.gz ...
Downloading http://yann.lecun.com/exdb/mnist/t10k-labels-idx1-ubyte.gz ...
0% |################################################################| 100%
Unzipped /home/davood/mnist/build/data/t10k-labels-idx1-ubyte.gz ...
-- Configuring done
-- Generating done
-- Build files have been written to: /home/davood/mnist/build

**Comment by zou3519** [comment]
cc @goldsborough

**Comment by goldsborough** [comment]
Is it possible that your release build is outdated? The necessary changes for this landed a few days ago. If you grab the newest version from https://download.pytorch.org/libtorch/nightly/cpu/libtorch-shared-with-deps-latest.zip, does that work for you? Please let me know.

**Comment by goldsborough** [comment]
The fix for that was in https://github.com/pytorch/pytorch/pull/13060. You need the latest release to get it

**Comment by dhaji** [comment]
I actually got this error after I installed the latest release you put in the above link

**Comment by dhaji** [comment]
I removed the previous folder and made a new one. However, still get the same error. Could you please share what you did step-by-step?

**Comment by goldsborough** [comment]
Ok I found the problem. I was on an outdated version of the mnist example. I had pushed a fix to the mnist example before the MNIST dataset fix to unsqueeze each image. After the MNIST dataset fix landed, this now resulted in images getting a 5th dimension, which is the cause of the error. I've updated the example so that it should now work (https://github.com/goldsborough/examples/commit/9a74ef4be3b1812f895ba2e2fe7a5ac9b1bdf07a)

Hope it'll now work.
